test/functional/tests/fault_injection/test_fault_injection_many_to_one.py

```diff
@@ -62,3 +62,117 @@ def test_remove_multilevel_core():
+        blocks = int(cache.size.value / block_size.value)
+        if cache_mode == CacheModeTrait.InsertRead:
+            dd = (Dd()
+                  .input(core1.system_path)
+                  .output("/dev/null")
+                  .block_size(block_size)
+                  .count(blocks))
+            dd.run()
+        elif cache_mode != CacheModeTrait.InsertRead:
+            dd = (Dd()
+                  .input("/dev/urandom")
+                  .output(core1.system_path)
+                  .block_size(block_size)
+                  .count(blocks))
+            dd.run()
+        core1_occupied_blocks = core1.get_occupancy()
+        occupied_blocks_before = cache.get_occupancy()
+
+    with TestRun.step("Remove the first core."):
+        cache.remove_core(core1.core_id, True)
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != 1:
+            TestRun.fail(f"Expected cores count: 1; Actual cores count: {cores_count}.")
+
+    with TestRun.step("Check if occupancy from the first core is removed from cache."):
+        occupied_blocks_after = cache.get_occupancy()
+        # Blocks occupied by the first cache should be completely released.
+        if occupied_blocks_after != occupied_blocks_before - core1_occupied_blocks:
+            TestRun.LOGGER.error("Blocks previously occupied by the first core "
+                                 "aren't released by removing this core.")
+
+    with TestRun.step("Check if the remaining core is able to use cache."):
+        try:
+            if cache_mode == CacheModeTrait.InsertRead:
+                dd = (Dd()
+                      .input(core2.system_path)
+                      .output("/dev/null")
+                      .block_size(block_size)
+                      .count(blocks_100MiB))
+                dd.run()
+                os_utils.sync()
+                os_utils.drop_caches()
+                dd = (Dd()
+                      .input(core2.system_path)
+                      .output("/dev/null")
+                      .block_size(block_size)
+                      .count(blocks_100MiB))
+                dd.run()
+            elif cache_mode != CacheModeTrait.InsertRead:
+                dd = (Dd()
+                      .input("/dev/urandom")
+                      .output(core2.system_path)
+                      .block_size(block_size)
+                      .count(blocks_100MiB))
+                dd.run()
+                os_utils.sync()
+                os_utils.drop_caches()
+                dd = (Dd()
+                      .input("/dev/urandom")
+                      .output(core2.system_path)
+                      .block_size(block_size)
+                      .count(blocks_100MiB))
+                dd.run()
```

Comment:
You could leave preparing dd command inside if block, and execute rest of code unconditional.